programs/bubblegum/program/src/processor/finalize_tree_with_root.rs

```diff
@@ -179,7 +187,13 @@ pub(crate) fn check_stake<'info>(
         &voter.voter_authority == staker_acc.key,
         BubblegumError::StakingVoterAuthorityMismatch
     );
-
+    // todo: use a non mutable version of the mining
+    let mut mining_data = mining_acc.data.borrow_mut();
```

Comment:
What if we make it like this?
Not to borrow mut.
Or even without convertion to account info?
```suggestion
    let mut mining_data = mining_acc.to_account_info().data.borrow();
```

Reply:
Have you tried to compile it? Afaik it requires mutable reference since the WrappedMining data structure has two mutable references under the hood, so I have doubts that `borrow()` is an option here.

Reply:
Wdym `without conversion to account info`? `mining_acc` _is_ account info, isn't it?